internal/decryption/decryption.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package decryption
+
+import (
+	"io"
+	"strings"
+
+	"github.com/mongodb/mongocli/internal/decryption/keyproviders"
+)
+
+type DecryptConfig struct {
+	lek             []byte
+	compressionMode CompressionMode
+}
+
+// Decrypt decrypts the content of an audit log file using the metadata found in the file,
+// the credentials provided by the user and the AES-GCM algorithm.
+// The decrypted audit log records are saved in the out stream.
+func Decrypt(logReader io.Reader, out io.Writer, credentialsProvider keyproviders.CredentialsProvider) error {
+	auditLogFormat, logLines, err := readAuditLogFile(logReader)
+	if err != nil {
+		return err
+	}
+
+	auditLogEncoding := newAuditLogEncoding(auditLogFormat)
+	output := buildOutput(out)
+	var decryptConfig *DecryptConfig
+	var logRecordIdx uint64
+
+	for idx, line := range logLines {
+		lineNb := idx + 1
+		if len(line) > 0 {
+			logLine, err := auditLogEncoding.Parse(line)
+			if err != nil {
```

Comment:
[nit] can we reduce the nesting here?
```suggestion
		if len(line) == 0 {
			continue
		}
		logLine, err := auditLogEncoding.Parse(line)
		if err != nil {
```

Reply:
done